components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -268,4 +356,84 @@ private Action updateEndpointAuthenticationProperties(String actionType, String
         return CACHE_BACKED_DAO.updateActionEndpointAuthProperties(actionType, actionId, authentication,
                 IdentityTenantUtil.getTenantId(tenantDomain));
     }
+
+    /**
+     * Perform pre validations on action model when creating an action.
+     *
+     * @param actionType Action type.
+     * @param action     Action create model.
+     * @throws ActionMgtException if action model is invalid.
+     */
+    private void doPreAddActionValidations(String actionType, Action action) throws ActionMgtException {
+
+        if (actionType.equals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType())) {
```

Comment:
We don't need to be specific on the action type as these fields are common.

Reply:
Updated with b22349fb9192bbbd7f2cf26535baa690d138d6f0